lambdas/process-journey-event/src/test/java/uk/gov/di/ipv/core/processjourneyevent/JourneyMapTest.java

```diff
@@ -122,6 +122,38 @@ void shouldMatchNestedJourneyEntryEvents(IpvJourneyTypes journeyType) throws IOE
         }
     }
 
+    @ParameterizedTest
+    @EnumSource
+    void basicEventTargetStatesShouldExist(IpvJourneyTypes journeyType) throws IOException {
+        var stateMachine = new StateMachineInitializer(journeyType).initialize();
+        var stateMachineKeys = stateMachine.keySet();
+
+        for (var targetKey : stateMachineKeys) {
+            var targetState = stateMachine.get(targetKey);
+            if (targetState instanceof BasicState basicState) {
+                var events = basicState.getEvents();
+                for (var event : events.values()) {
+                    if (event instanceof BasicEvent basicEvent) {
+                        if (basicEvent.getTargetJourney() != null) {
+                            var basicEventStateMachine =
+                                    new StateMachineInitializer(
+                                                    IpvJourneyTypes.valueOf(
+                                                            basicEvent.getTargetJourney()))
+                                            .initialize();
+                            var basicEventStateMachineKeys = basicEventStateMachine.keySet();
+                            assertTrue(
+                                    basicEventStateMachineKeys.contains(
+                                            basicEvent.getTargetState()));
```

Comment:
What's the error message if this fails? Might be worth passing a custom message if there's no more suitable assertion to use, e.g.:
```
assertTrue(stateMachineKeys.contains(basicEvent.getTargetState()),
        "Unknown target state %s".formatted(basicEvent.getTargetState()));
```

Reply:
We just get an assertion failed message - will add this :)